test/AddressDerivation.js

```diff
@@ -90,17 +85,18 @@ if (isHardhat) {
                 const expectedTokenManagerAddress = '0x89AF99D9373722De5F29ABbF706efeD020ae3E1F';
 
                 const params = defaultAbiCoder.encode(['bytes', 'address'], [operator, expectedTokenAddress]);
-                const payload = defaultAbiCoder.encode(
-                    ['uint256', 'bytes32', 'string', 'string', 'uint8', 'bytes'],
-                    [MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN, tokenId, tokenName, tokenSymbol, tokenDecimals, minter],
-                );
-                const wrappedPayload = defaultAbiCoder.encode(
-                    ['uint256', 'string', 'bytes'],
-                    [MESSAGE_TYPE_RECEIVE_FROM_HUB, sourceChain, payload],
+                const { payload } = encodeDeployInterchainToken(
+                    MESSAGE_TYPE_RECEIVE_FROM_HUB,
```

Comment:
This argument is out of place for what the function is trying to do. Look at how we have structured the ItsMessage vs ItsHubMessage in stellar ITS. Similar I recommend having two layers, encode* function for the underlying ITS message and then another encode method for send to hub or receive from hub